diff --git a/src/jemalloc.c b/src/jemalloc.c
index ed13718..bd21074 100644
--- a/src/jemalloc.c
+++ b/src/jemalloc.c
@@ -1032,6 +1032,8 @@ malloc_conf_init_helper(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS],
 	const char *opts, *k, *v;
 	size_t klen, vlen;
 
+	printf("we force set jemalloc profiling prof:true!!!\n");
+	opt_prof=true;
 	for (i = 0; i < MALLOC_CONF_NSOURCES; i++) {
 		/* Get runtime configuration. */
 		if (initial_call) {
